src/input/touch.c

```diff
@@ -100,9 +106,14 @@ touch_down(struct wl_listener *listener, void *data)
 	double sx = lx - x_offset;
 	double sy = ly - y_offset;
 
-	if (surface) {
-		wlr_seat_touch_notify_down(seat->seat, surface,
+	if (touch_point->surface) {
```

Comment:
The calculation for `sx`, `sy` is only needed within this if statement, so we might move the calculation for `lx`, `ly`, `sx` and `sy` into this condition.

Reply:
Will do.

Reply:
Done...

Reply:
Those lines can also go into the branch
```c
double lx, ly;
wlr_cursor_absolute_to_layout_coords(seat->cursor, &event->touch->base,
		event->x, event->y, &lx, &ly);
```